onboarding_payload_test_suite/onboarding_script_support.py

```diff
@@ -172,9 +172,9 @@ def payload_version_check(self, version: int) -> None:
         logger.info("Verified QR code payload version: {bin(version)}")
 
     def payload_rendezvous_capabilities_bit_mask_check(
-        self, discovery_capabilities_bitmask: int
+        self, discovery_capabilities_bitmask: int, max_bit: int
     ) -> None:
-        if not (0 <= discovery_capabilities_bitmask <= 7):
+        if not (0 <= discovery_capabilities_bitmask <= max_bit):
```

Comment:
```suggestion
        DISCOVERY_CAP_BLE = (1 << 1)
        DISCOVERY_CAP_IP = (1 << 2)
        DISCOVERY_CAP_PAF = (1 << 3)                
        ALL_CAPABILITIES_MASK = DISCOVERY_CAP_BLE | DISCOVERY_CAP_IP | DISCOVERY_CAP_PAF
        if ((discovery_capabilities_bitmask & ~ALL_CAPABILITIES_MASK) != 0) or ((discovery_capabilities_bitmask & ALL_CAPABILITIES_MASK) == 0):
```

This can then easily be expanded to support NFC transport layer which is bit (1 << 4) coming in 1.5.

This also makes sure that there is at least one bit set, and that no bits outside the mask are set.